src/cli.ts

```diff
@@ -1,51 +1,537 @@
+      group: 'Platform Specific Options',
+      type: 'boolean',
+    })
+    .option('build-version', {
+      description:
+        '(macOS, windows only) The build version of the app. Maps to `FileVersion` metadata property on Windows, and `CFBundleVersion` on macOS',
+      group: 'Platform Specific Options',
+      type: 'string',
+    })
+    .option('counter', {
+      default: false,
+      description:
+        '(macOS only) set a dock count badge, determined by looking for a number in the window title',
+      group: 'Platform Specific Options',
+      type: 'boolean',
+    })
+    .option('darwin-dark-mode-support', {
+      default: false,
+      description: '(macOS only) enable Dark Mode support on macOS 10.14+',
+      group: 'Platform Specific Options',
+      type: 'boolean',
+    })
+    .option('f', {
+      alias: 'fast-quit',
+      default: false,
+      description: '(macOS only) quit app on window close',
+      group: 'Platform Specific Options',
+      type: 'boolean',
+    })
+    .option('title-bar-style', {
+      choices: ['hidden', 'hiddenInset'],
+      description:
+        '(macOS only) set title bar style. Consider injecting custom CSS (via --inject) for better integration',
+      group: 'Platform Specific Options',
+      type: 'string',
+    })
+    .option('win32metadata', {
+      coerce: parseJson,
+      description:
+        '(windows only) a JSON string of key/value pairs (ProductName, InternalName, FileDescription) to embed as executable metadata',
+      group: 'Platform Specific Options',
+      type: 'string',
+    })
+    // Debug Options
+    .option('crash-reporter', {
+      description: 'remote server URL to send crash reports',
+      group: 'Debug Options',
+      type: 'string',
+    })
+    .option('verbose', {
+      default: false,
+      description: 'enable verbose/debug/troubleshooting logs',
+      group: 'Debug Options',
+      type: 'boolean',
+    })
+    .version()
+    .help()
+    .group(['version', 'help'], 'Other Options')
+    .wrap(yargs.terminalWidth());
 
-// package.json is `require`d to let tsc strip the `src` folder by determining
-// baseUrl=src. A static import would prevent that and cause an ugly extra "src" folder
-const packageJson = require('../package.json'); // eslint-disable-line @typescript-eslint/no-var-requires
+  // Have to access argv to get yargs to actually process the args passed to it
+  // Do this now to go ahead and get any errors out of the way
+  args.argv;
 
-function collect(val: any, memo: any[]): any[] {
-  memo.push(val);
-  return memo;
+  return args;
 }
 
-function getProcessEnvs(val: string): any {
-  if (!val) {
-    return {};
-  }
-  return parseJson(val);
-}
+export function parseArgs(args: yargs.Argv<any>): any {
+  const parsed = { ...args.argv };
+  parsed.targetUrl = parsed._.length > 0 ? parsed._[0].toString() : '';
+  parsed.out = parsed._.length > 1 ? parsed._[1] : '';
```

Comment:
What is `parsed._` in yargsland? A comment / link to docs will be welcome.

Reply:
Will do, though for reference it's how it handles positional arguments (such as targetUrl and outputDirectory)